### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by AlexanderZaborski-tomtom**

### CODE REVIEW

File: custom_endpoint/src/Controller/CustomEndpointController.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ */
+class CustomEndpointController extends ControllerBase {
+
+  /**
+   * Custom Endpoint controller constructor
+   *
+   * @param \Drupal\Core\Session\AccountInterface
+   */
+  public function __construct(AccountInterface $currentUser) {
+    $this->currentUser = $currentUser;
```
Comment (AdamBodera-TomTom) [inline custom_endpoint/src/Controller/CustomEndpointController.php]: now we don't use a current user here so remove this
instead we want to create a class member (field) that will store the injected current_user_service

File: custom_endpoint/src/Controller/CustomEndpointController.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    );
+  }
+
+  /**
+   * @return JsonResponse with user data
+   */
+  public function renderApi(): JsonResponse {
+
+    return new JsonResponse([
+      'data' => $this->currentUserService->getResults($this->currentUser),
```
Comment (AdamBodera-TomTom) [inline custom_endpoint/src/Controller/CustomEndpointController.php]: you already called it here, so you need to declare it in the class
and we don't pass the currentUser to the service, the service will have it there

File: custom_endpoint/src/CurrentUserService.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Drupal\custom_endpoint;
+
+class CurrentUserService {
+
```
Comment (AdamBodera-TomTom) [inline custom_endpoint/src/CurrentUserService.php:11]: your service now needs a constructor and a current_user field

File: custom_endpoint/src/Controller/CustomEndpointController.php [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+use Drupal\Core\Session\AccountInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\custom_endpoint\CurrentUserService;
+
+/**
+ * Custom Endpoint Controller class
+ */
+class CustomEndpointController extends ControllerBase {
+
+  public $currentUserService;
```
Comment (AdamBodera-TomTom) [inline custom_endpoint/src/Controller/CustomEndpointController.php]: always use the lowest possible access level in class
in this case private

File: custom_endpoint/custom_endpoint.info.yml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+name: Custom user endpoint
+description: Endpoint that outputs user data
+
+type: module
+core_version_requirement: ^8 || ^9
```
Comment (AdamBodera-TomTom) [inline custom_endpoint/custom_endpoint.info.yml]: we are developing in Drupal 9 :)

File: custom_endpoint/custom_endpoint.routing.yml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+custom_endpoint.jsonapi:
+  path: '/api/currentuser'
+  defaults:
+    _controller: '\Drupal\custom_endpoint\Controller\CustomEndpointController::renderApi'
+    _title: 'Current user data'
+  methods: ['GET']
+  requirements:
+    _access: 'TRUE'
```
Comment (AdamBodera-TomTom) [inline custom_endpoint/custom_endpoint.routing.yml]: all your files should end with one empty line

File: custom_endpoint/src/Controller/CustomEndpointController.php [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
```
Comment (AdamBodera-TomTom) [inline custom_endpoint/src/Controller/CustomEndpointController.php]: missing class member comment block

File: custom_endpoint/src/Controller/CustomEndpointController.php [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+  /**
+   * Custom Endpoint controller constructor
+   *
+   * @param Drupal\custom_endpoint\CurrentUserService
```
Comment (AdamBodera-TomTom) [inline custom_endpoint/src/Controller/CustomEndpointController.php]: the FQCN needs to start with the trailing `\` and point to the parameter which it defines, also the each parameter needs a description, so in this case:
```
* @param \Drupal\custom_endpoint\CurrentUserService $currentUser
*   The current user service.
```

File: custom_endpoint/src/Controller/CustomEndpointController.php [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+   */
+  public function __construct(CurrentUserService $currentUser) {
```
Comment (AdamBodera-TomTom) [inline custom_endpoint/src/Controller/CustomEndpointController.php]: parameters and variables in Drupal are snake-cased, so in this case:
`$current_user`

only class members are camel-cased

File: custom_endpoint/src/CurrentUserService.php [hunk 10]
```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Drupal\custom_endpoint;
+
+use Drupal\Core\Session\AccountInterface;
+
+class CurrentUserService {
```
Comment (AdamBodera-TomTom) [inline custom_endpoint/src/CurrentUserService.php:10]: missing class description

File: custom_endpoint/src/CurrentUserService.php [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+
+  /**
+   * Return current user data
+   * or return an error message.
+   */
+  public function getResults(): array {
+    //Try to get user data
+    try {
+      $username = $this->currentUser->getAccountName();
+      $created = $this->currentUser->get['created'];
```
Comment (AdamBodera-TomTom) [inline custom_endpoint/src/CurrentUserService.php]: instead use:
`$this->currentUser->get('created')->value`
or
`$this->currentUser->get('created')->getValue()`

File: custom_endpoint/src/CurrentUserService.php [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+      $moddate = $this->currentUser->getLastAccessedTime();
+      $lastlogon = $this->currentUser->getLastAccessedTime();
+
+      if ($username == "") {
```
Comment (AdamBodera-TomTom) [inline custom_endpoint/src/CurrentUserService.php]: this is such a weak check :D
use routing file to limit access to your endpoint

File: custom_endpoint/src/CurrentUserService.php [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+        return "You are not logged in.";
```
Comment (AdamBodera-TomTom) [inline custom_endpoint/src/CurrentUserService.php]: in Drupal string literals should be delimited by single quotes

Comment (AlexanderZaborski-tomtom) [reply custom_endpoint/src/CurrentUserService.php]: Fixed

File: custom_endpoint/src/CurrentUserService.php (hunk 11 above)
Comment (AlexanderZaborski-tomtom) [reply custom_endpoint/src/CurrentUserService.php]: Added ->value

For some reasion getValue() didn't work.

File: custom_endpoint/src/Controller/CustomEndpointController.php (hunk 4 above)
Comment (AlexanderZaborski-tomtom) [reply custom_endpoint/src/Controller/CustomEndpointController.php]: done

File: custom_endpoint/custom_endpoint.routing.yml (hunk 6 above)
Comment (AlexanderZaborski-tomtom) [reply custom_endpoint/custom_endpoint.routing.yml]: Done

File: custom_endpoint/custom_endpoint.info.yml (hunk 5 above)
Comment (AlexanderZaborski-tomtom) [reply custom_endpoint/custom_endpoint.info.yml]: Fixed

File: custom_endpoint/src/CurrentUserService.php (hunk 12 above)
Comment (AlexanderZaborski-tomtom) [reply custom_endpoint/src/CurrentUserService.php]: fixed

File: custom_endpoint/src/CurrentUserService.php (hunk 10 above)
Comment (AlexanderZaborski-tomtom) [reply custom_endpoint/src/CurrentUserService.php:10]: done

File: custom_endpoint/src/Controller/CustomEndpointController.php (hunk 9 above)
Comment (AlexanderZaborski-tomtom) [reply custom_endpoint/src/Controller/CustomEndpointController.php]: fixed

File: custom_endpoint/src/Controller/CustomEndpointController.php (hunk 8 above)
Comment (AlexanderZaborski-tomtom) [reply custom_endpoint/src/Controller/CustomEndpointController.php]: fixed

File: custom_endpoint/src/Controller/CustomEndpointController.php (hunk 7 above)
Comment (AlexanderZaborski-tomtom) [reply custom_endpoint/src/Controller/CustomEndpointController.php]: Added

File: custom_endpoint/src/Controller/CustomEndpointController.php [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+  public function __construct(CurrentUserService $current_user) {
+    $this->currentUserService = $current_user;
+  }
+
+  /**
+   * Gets the current user service
+   *
+   * @param \Symfony\Component\DependencyInjection\ContainerInterface;
+   */
+  public static function create(ContainerInterface $container): CustomEndpointController {
```
Comment (AdamBodera-TomTom) [inline custom_endpoint/src/Controller/CustomEndpointController.php:36]: the comment block pattern looks like this:

```
  /**
   * Returns the current user's profile data.
   *
   * @param \Symfony\Component\HttpFoundation\Request $request
   *   The request object.
   *
   * @return \Symfony\Component\HttpFoundation\Response
   *   The response.
   */
  public function getUserProfileData(Request $request): Response {
```

pay attention to indentations

**Pull request merged**